src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -72,9 +70,9 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
       .write
       .insertInto("filesSource")
 
-    val dfWithTestDirectory = retryWhile[Array[Row]](
-      spark.sql(s"select * from filesSource where assetIds = Array(1739842716040355)").collect,
-      rows => rows.length < 10)
+    val dfWithTestDirectory = retryWhile[List[File]](
+      writeClient.files.filter(FilesFilter(assetIds = Some(Seq(1739842716040355L)))).compile.toList,
+      files => files.length != 10)
```

Comment:
"< 10" is better, because if someone creates a file matching this at least the test will fail sooner on the next line instead of waiting for the full timeout.